assets/src/components/mapPage/routePropertiesCard.tsx

```diff
@@ -0,0 +1,259 @@
+import React, { useState } from "react"
+import { useRoute } from "../../contexts/routesContext"
+import { CircleCheckIcon } from "../../helpers/icon"
+import {
+  RoutePattern,
+  RoutePatternId,
+  DirectionName,
+  DirectionId,
+  RouteId,
+  Route,
+  ByRoutePatternId,
+} from "../../schedule"
+import { Card } from "../card"
+import CloseButton from "../closeButton"
+import { RoutePill } from "../routePill"
+
+const sortRoutePatterns = (routePatterns: RoutePattern[]): RoutePattern[] =>
+  routePatterns.sort((rp1, rp2) => rp1.sortOrder - rp2.sortOrder)
+
+const displayName = (routePattern: RoutePattern) => {
+  //const splitName = routePattern.name.split("- ", 2)
+  return routePattern.name
+}
+
+const DetailSection = ({
+  title,
+  isOpen,
+  toggleOpen,
+  className,
+  children,
+}: {
+  title: string
+  isOpen: boolean
+  toggleOpen: () => void
+  className: string
+  children: JSX.Element
+}) => {
+  return (
+    <details
+      className={`m-route-properties-card__details-section ${className}`}
+      {...(isOpen ? { open: true } : {})}
+    >
```

Comment:
question/suggestion: what is the reasoning behind the object spread here?

Does this maybe work here?

```suggestion
open={isOpen}
```

This might be closer to the current code
```suggestion
open={isOpen || undefined}
```

Reply:
Yes, this is much nicer, thanks! I was thinking we had to leave off the `open` attribute entirely when it is not open, but setting to `undefined` is better!

Reply:
The reason  `open={isOpen}` doesn't work is because the [attribute is a boolean ](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details#attr-open), which tripped me up for a bit!

Reply:
I learned[^1] that jsx properly handles the `false` state on [boolean attributes](https://developer.mozilla.org/en-US/docs/Glossary/Boolean) , so you might try leaving off the `|| undefined` part even.

[^1]: this attribute worked with `boolean` even though it was supposed to be left off if `hidden=false` https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/hidden so I've learned that default html props in jsx are not always 1 to 1 with the actual resulting html output

Reply:
> The reason `open={isOpen}` doesn't work is because the [attribute is a boolean ](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details#attr-open), which tripped me up for a bit!

Oh interesting, maybe it's inconsistent or requires jsx to implement extra handling on top of how it renders html. `hidden` isn't a `boolean` attribute but I expected that `boolean` attributes might work.

Reply:
Ah you know what there must have been something else at play when this first wasn't working for me - trying again with `open={isOpen}` works!